internal/gcs/process.go

```diff
@@ -123,26 +127,53 @@ func (gc *GuestConnection) exec(ctx context.Context, cid string, params interfac
 
 // Close releases resources associated with the process and closes the
 // associated standard IO streams.
-func (p *Process) Close() error {
+func (p *Process) Close() (err error) {
 	ctx, span := oc.StartSpan(context.Background(), "gcs::Process::Close")
 	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
 	span.AddAttributes(
 		trace.StringAttribute("cid", p.cid),
 		trace.Int64Attribute("pid", int64(p.id)))
 
-	if err := p.stdin.Close(); err != nil {
-		log.G(ctx).WithError(err).Warn("close stdin failed")
+	if p.gcClosed() {
+		return hcs.ErrAlreadyClosed
 	}
-	if err := p.stdout.Close(); err != nil {
-		log.G(ctx).WithError(err).Warn("close stdout failed")
+
+	p.stdioLock.Lock()
+	if p.stdin != nil {
```

Comment:
Why are we not just using the Process funcs for "CloseStd*"?

Reply:
It would involve creating multiple spans, and each call would have to lock and unlock `stdioLock`, which seems wasteful.